### PR TITLE
MODINVSTOR-625 - Change effective location for items if parent holding location is changed

### PR DESCRIPTION
**Pull request by siarhei-charniak**
[MODINVSTOR-625](https://issues.folio.org/browse/MODINVSTOR-625) - Change effective location for items if parent holding location is changed
Added functionality to update item effective location upon holding or item update. 
Removed legacy database triggers.

### CODE REVIEW

File: src/main/java/org/folio/rest/impl/HoldingsStorageAPI.java [hunk 1]
```diff
@@ -307,7 +308,7 @@ public void putHoldingsStorageHoldingsByHoldingsRecordId(
                   if (Objects.equals(entity.getHrid(), existingHoldings.getHrid())) {
                     try {
                       postgresClient.startTx(connection -> {
-                        updateItemEffectiveCallNumbersByHoldings(connection, postgresClient, entity).onComplete(updateResult -> {
+                        updateItemEffectiveCallNumbersAndLocationsByHoldings(connection, postgresClient, entity).onComplete(updateResult -> {
```
Comment (MikhailFokanov) [inline src/main/java/org/folio/rest/impl/HoldingsStorageAPI.java:311]: I'd change it to updateRelatedItemsAttributes for brevity.

**Review by MikhailFokanov — APPROVED** [review]
@marc-johnson, should we wait for your approval?

**Comment by marcjohnson-kint** [comment]
> @marc-johnson, should we wait for your approval?

There is no need to wait for my approval. The teams are responsible for their own work. If you are acting as a reviewer for that team, then no further reviews are necessary.

File: src/main/java/org/folio/rest/impl/HoldingsStorageAPI.java (hunk 1 above)
Comment (marcjohnson-kint) [reply src/main/java/org/folio/rest/impl/HoldingsStorageAPI.java]: I might propose `updateItemEffectiveProperties` or `deriveItemEffectiveProperties` instead.

**Pull request merged**